R/drug_response_core.R

```diff
@@ -0,0 +1,463 @@
+#' @import dplyr
+#' @importFrom stats median
+#' @importFrom dplyr bind_rows select mutate filter left_join group_by summarise
+#' @importFrom utils head
+#' @importFrom rlang .data
+NULL
+
+# Declare global variables to avoid R CMD check notes
+utils::globalVariables(c("FINNGENID", "EVENT_AGE", "first_drug_age", "time_to_drug"))
+
+#' @title data object returned from drug response analyse (create_drug_response)
+#' @param responses data frame with response data
+#' @param lab_measurements data frame with all lab measurements
+#' @param drug_purchases data frame with all drug purchases
+#' @param before_period vector with two elements, start and end of the before period # nolint: line_length_linter.
+#' @param after_period vector with two elements, start and end of the after period # nolint: line_length_linter.
+#' @return object of class drug.response
+#' @export
+
+drug.response <- function(responses, lab_measurements,
+                          drug_purchases,
+                          before_period, after_period) {
+  obj <- list(responses = responses,
+              all_measurements = lab_measurements,
+              all_drug_purchases = drug_purchases,
+              lab_response_period = list(before_period = before_period,
+                                         after_period = after_period))
+  class(obj) <- "drug.reponse"
+  obj
+}
+
+#' @title Create drug response object
+#' @param conn Connection object or kanta data frame (for legacy support)
+#' @param lablist Lab list or phenos data frame (for legacy support)
+#' @param druglist Drug list
+#' @param before_period Before period
+#' @param after_period After period
+#' @param finngen_ids Finngen IDs
+#' @param remove_outliers_sd Remove outliers SD
+#' @param covariates Covariates data frame
+#' @param covariate_cols Covariate columns
+#' @param all_labs Direct labs data frame (alternative input)
+#' @param all_phenos Direct phenos data frame (alternative input)
+#' @param kanta Legacy parameter name for labs data
+#' @param phenos Legacy parameter name for phenos data
+#' @return drug.response object
+#' @export
+create_drug_response <- function(conn = NULL, lablist = NULL, druglist = NULL,
+                                 before_period = NULL, after_period = NULL,
+                                 finngen_ids = NULL, remove_outliers_sd = NULL,
+                                 covariates = NULL,
+                                 covariate_cols = NULL,
+                                 # Alternative input parameters
+                                 all_labs = NULL, all_phenos = NULL,
+                                 # Legacy parameters
+                                 kanta = NULL, phenos = NULL) {
+
+  # Handle different input types
```

Comment:
I would really drop the legacy parameters here and just require fg_connect object and thats it. becomes too complex to handle quickly